util/cluster.go

```diff
@@ -0,0 +1,205 @@
+package util
+
+import (
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/giantswarm/columnize"
+	"github.com/giantswarm/gscliauth/config"
+	"github.com/giantswarm/gsclientgen/models"
+	"github.com/giantswarm/gsctl/client"
+	"github.com/giantswarm/gsctl/client/clienterror"
+	"github.com/giantswarm/gsctl/commands/errors"
+	"github.com/giantswarm/gsctl/confirm"
+	"github.com/giantswarm/microerror"
+	"github.com/spf13/afero"
+)
+
+const (
+	listClustersActivityName = "list-clusters"
+	clusterCacheFileName     = "clustercache"
+)
+
+// GetClusterID gets the cluster ID for a provided name/ID
+// by checking in both the user cache and on the API
+func GetClusterID(clusterNameOrID string, clientWrapper *client.Wrapper) (string, error) {
+	isID := IsInClusterCache(clusterNameOrID)
```

Comment:
"isID" implies that we can find out whether this is an ID. In fact, we can only find out if it is an ID we know. So `isKnownID` or `isCachedID` would fit better.